core/src/OpenACC/Kokkos_OpenACC_ParallelReduce_MDRange.hpp

```diff
@@ -84,7 +89,14 @@ class Kokkos::Impl::ParallelReduce<CombinedFunctorReducerType,
         m_policy);
 
     reducer.final(&val);
-    *m_result_ptr = val;
+    if (m_result_ptr_on_device == false) {
+      acc_wait(async_arg);
+      *m_result_ptr = val;
+    } else {
+      acc_memcpy_to_device_async(m_result_ptr, &val, sizeof(ValueType),
+                                 async_arg);
+      acc_wait(async_arg);
```

Comment:
If the result is device-accessible, `parallel_reduce` should not fence.
```suggestion
```

Reply:
The current OpenACC reduction implementation uses the same OpenACC implementation for both the host reduction variable and device reduction variable, and we need `acc_wait(async_arg)` here because `val` is a local variable.
In fact, we can have separate implementations for host reductions and device reductions, but the current implementation reuses the host reduction implementation for device reduction to avoid too lengthy code.

Reply:
Hmm... OK. If this needs to be a local variable, would you mind adding a comment to that effect?

Reply:
Added a comment.